### PR TITLE
Task-12: Add validation

### PR DESCRIPTION
**Pull request by vpuzan**
We choose to use such a wonderful library as Joi.

### CODE REVIEW

File: validation/validation.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Exception from "../exception/exception.js";
+
+export const validateRequest = (schema) => (req, res, next) => {
+    let error;
+    if (Object.keys(req.body).length === 0){
+        error = schema.validate(req.query);
```
Comment (mdziarkach) [inline validation/validation.js]: joi should throw exception if required keys are missing

File: validation/validation.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    } else{
+        error = schema.validate(req.body);
+    }
+    if (error.error) {
+        throw new Exception(400, error.error.stack);
```
Comment (mdziarkach) [inline validation/validation.js]:
```
next(new Exception...)
```

File: validation/validation.js (hunk 1 above)
Comment (mdziarkach) [reply validation/validation.js]: const { error } = req.method === 'GET' ? schema.validate(req.query) : schema.validate(req.body);

Comment (vpuzan) [reply validation/validation.js]: done

**Review by mdziarkach — APPROVED** [review]

**Comment by mdziarkach** [comment]
already approved :)

**Pull request merged**